sdk/lib/core/uri.dart

```diff
@@ -512,21 +554,38 @@ abstract interface class Uri {
 
   /// The fragment identifier component.
   ///
+  /// A fragment identifier is a string after URI, after the hash, which
```

Comment:
I'd go with something like:
```dart
/// A fragment identifier is the part of a URI after a `#` character.
/// It is intended to indentify something specific to the document.
/// In a link to a web page, the fragment traditionally identifies a section
/// or part of the page by its HTML `id` attribute.
/// In modern browsers, it can also identify a part of the page to be
/// recognized by the browser, for example a specific text to find and
/// show to the user.
```